### PR TITLE
feat(Tile): Add actionable prop

### PR DESCRIPTION
**Pull request by veekeys**
Added new prop to consume `iui-actionable` class.
Prop name and description up for debate.
Added all div props, so users could add `onClick` handler for example for interactable tile.
New story shows weird new Tile behavior, when onClick it gets `selected` state, but when you unselect, thumbnail is still scaled, because we have also `focus-within` 
image

Better to wait for #630 to be merged (or ignore other than Tile changes)

### CODE REVIEW

File: stories/core/Tile.stories.tsx [hunk 1]
```diff
@@ -131,6 +133,68 @@ AllProps.argTypes = { ...Basic.argTypes };
 AllProps.args = {
   ...Basic.args,
   isSelected: true,
+  isActionable: true,
```
Comment (bentleyvk) [inline stories/core/Tile.stories.tsx:136]: Why passing like that? It is not visible to the users how props look.

Comment (veekeys) [reply stories/core/Tile.stories.tsx:136]: What do you mean its not? I also destructure in the code itself and assign default value

File: src/core/Tile/Tile.test.tsx [hunk 2]
```diff
@@ -221,3 +221,8 @@ it('should propagate misc props correctly', () => {
   expect(child).toBeTruthy();
   expect(child.textContent).toBe('test-content');
 });
+
+it('should render actionable tile', () => {
```
Comment (adamhock) [inline src/core/Tile/Tile.test.tsx:225]: Should we write a test for tabbing to an actionable tile?

Comment (veekeys) [reply src/core/Tile/Tile.test.tsx:225]: And what would you like me to test there?

File: stories/core/Tile.stories.tsx [hunk 3]
```diff
@@ -131,6 +133,68 @@ AllProps.argTypes = { ...Basic.argTypes };
...
+      thumbnail={thumbnail}
+      buttons={buttons}
+      leftIcon={leftIcon}
+      rightIcon={rightIcon}
+      isSelected={selected}
+      isNew={isNew}
+      moreOptions={moreOptions}
+      isActionable={isActionable}
+      {...rest}
+      onClick={() => setSelected((prev) => !prev)}
```
[the remark this answers is not in the document]
Comment (veekeys) [reply stories/core/Tile.stories.tsx:189]: I dont know.. It is not a custom action we have added (which we mostly want to show off in demos), but more of a funcnioliaty which was not there before (so actionable and onclick handler was not possible to be used straight on Tile).
`onClick` here is just same native handler as anything else available on dom elements.

File: src/core/Tile/Tile.test.tsx (hunk 2 above)
Comment (adamhock) [reply src/core/Tile/Tile.test.tsx:225]: I was thinking appending this test with this

`expect(actionTile.getAttribute('tabIndex')).toBe('0');`

I guess I don't see other components have a test for this, so I guess we're good here

**Review by adamhock — APPROVED** [review]

**Review by bentleyvk — APPROVED** [review]

**Pull request merged**